### PR TITLE
Show filter text as user changes control values

### LINKED ISSUE #529: Show filter text as user changes control values
I think it would be useful(especially to new users) to show the text of a filter as it would appear if it was applied and shown in the context panel as the user changes the control. So, for example, if I set the upper bound of a range control for Age to `3.0`, this "preview" would say `Age is less than or equal to 3.0`. I think this would help to explain the controls to the user in plain english and show them what their changes will effect.

### PR DESCRIPTION
**Pull request by hassanNS**
Fix #529 
single selection
genotype
multi selection
functclas

null
hgmd

The range one is left aligned as Don suggested.

Here is search
cohort

Here is infograph on openmrs
sdgt

The others are never used. 

Signed-off-by: Sheik Hassan <solergiga@yahoo.com>

### CODE REVIEW

**Comment by hassanNS** [comment]
This should be all good to go

**Comment by bruth** [comment]
Thanks @hassanNS. I will take a look at it today or tomorrow.

**Comment by naegelyd** [comment]
@hassanNS Without thoroughly reviewing this whole PR again yet, I'm wondering if more could be done to increase the reusability of the work done here. I see a couple possible approaches:

1. Use the server to generate the language. As it stands now, the server generates the language and includes it in the response when applying a filter and then that is [flattened](https://github.com/chop-dbhi/cilantro/blob/2ccc1aa1bee066ab90028a64c7af88d6438e1237/src/js/cilantro/ui/context/filters.js#L11) and displayed. While this approach would probably be pretty simple, it may result in a lot of API calls. I personally have no reason to think these would take very long since it would only be generating a small, single filter, context response but there may be other ramifications I have not yet considered.

2. Modify the base control with a `getLanguage` method like it already has the [getValue](https://github.com/chop-dbhi/cilantro/blob/2ccc1aa1bee066ab90028a64c7af88d6438e1237/src/js/cilantro/ui/controls/base.js#L195), [getOperator](https://github.com/chop-dbhi/cilantro/blob/2ccc1aa1bee066ab90028a64c7af88d6438e1237/src/js/cilantro/ui/controls/base.js#L194), etc. This then, would be defined once one the base control but would allow custom controls to override it. The language construction should be pretty universal across all the controls in Cilantro proper. Basically, consider the operator and then format with values. I think all this logic could be incorporated into a single `getLanguage()` method which could like on the [base control](https://github.com/chop-dbhi/cilantro/blob/master/src/js/cilantro/ui/controls/base.js) and then this PR wouldn't change all the controls but would instead just change a single control that all inherit from. This leads to how the language will be shown. I believe the base control template([container](https://github.com/chop-dbhi/cilantro/blob/master/src/templates/field/form-condensed.html#L7) and [ui](https://github.com/chop-dbhi/cilantro/blob/36e0df2fb60027b2f596b76624a6f1e8d40c8495/src/js/cilantro/ui/field/form.js#L65)) could be updated to include an element for the language and that language could be reset when the `change` event occurs. Individual controls could then also hide the language by making `getLanguage` return the empty string __or__ by toggling a setting on the base control that determines whether or not to use the language.

So, to conclude, I think `1` might require the least to change on client side and would guarantee language consistency between this update and the context panel on the right. `2` has the advantage of not having to make server calls but would make a significant change to the base control. Either way, I think the work here is good but I think it is fairly redundant and could be condensed. I probably should have done a better job outlining these approaches when I drew up this ticket originally...

@bruth Thoughts on approaches above?

**Comment by aaron0browne** [comment]
Can this be closed?
